cpp/tests/io/parquet_test.cpp

```diff
@@ -4398,20 +4409,31 @@ TEST_F(ParquetReaderTest, NestedByteArray)
 
   auto const expected = table_view{{int_col, float_col, list_list_int_col}};
   EXPECT_EQ(3, expected.num_columns());
-  cudf_io::table_input_metadata ouput_metadata(expected);
-  ouput_metadata.column_metadata[0].set_name("col_other");
-  ouput_metadata.column_metadata[1].set_name("col_float");
-  ouput_metadata.column_metadata[2].set_name("col_binary").set_output_as_binary(true);
+  cudf_io::table_input_metadata output_metadata(expected);
+  output_metadata.column_metadata[0].set_name("col_other");
+  output_metadata.column_metadata[1].set_name("col_float");
+  output_metadata.column_metadata[2].set_name("col_binary").child(1).set_output_as_binary(true);
 
   auto filepath = temp_env->get_temp_filepath("NestedByteArray.parquet");
   cudf_io::parquet_writer_options out_opts =
     cudf_io::parquet_writer_options::builder(cudf_io::sink_info{filepath}, expected)
-      .metadata(&ouput_metadata);
+      .metadata(&output_metadata);
   cudf_io::write_parquet(out_opts);
 
+  auto source = cudf_io::datasource::create(filepath);
+  cudf_io::parquet::FileMetaData fmd;
+
+  read_footer(source, &fmd);
+  EXPECT_EQ(fmd.schema[5].type, cudf::io::parquet::Type::BYTE_ARRAY);
+
+  std::vector<cudf_io::reader_column_schema> md{
+    {},
+    {},
+    cudf_io::reader_column_schema().add_child(
+      cudf_io::reader_column_schema().set_convert_binary_to_strings(false))};
```

Comment:
This would be less verbose if the `reader_column_schema()` constructor took a `bool` argument. Are we avoiding doing this because we might add more properties to `reader_column_schema` in the future?

Reply:
I was, but I'm not against doing it.